### PR TITLE
starfetch: init at 0.0.2

### PR DESCRIPTION
**Pull request by auroraanna**
###### Description of changes

Add the starfetch package

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by papojari** [comment]
Whoops, my automatic nix-fmt on save did a lot of unwanted things to `pkgs/top-level/all-packages.nix`.

**Comment by papojari** [comment]
Result of `nixpkgs-review pr 167541` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>starfetch</li>
  </ul>
</details>

**Comment by papojari** [comment]
Added `meta.homepage`.

**Comment by papojari** [comment]
Updated to newest commit.

File: pkgs/tools/misc/starfetch/default.nix
Comment (06kellyjac) [inline pkgs/tools/misc/starfetch/default.nix:22]: Maybe double check this is Only and not Plus

https://github.com/Haruno19/starfetch/blob/main/LICENSE#L640

File: pkgs/tools/misc/starfetch/default.nix
Comment (06kellyjac) [inline pkgs/tools/misc/starfetch/default.nix:14]: Needs pre and post hooks

File: pkgs/tools/misc/starfetch/default.nix
Comment (06kellyjac) [inline pkgs/tools/misc/starfetch/default.nix:24]: Should add platforms

File: pkgs/tools/misc/starfetch/default.nix
Comment (06kellyjac) [inline pkgs/tools/misc/starfetch/default.nix:5]: Since the maintainer is interested in this being packaged id advise they tag a version 0.0.1
People are more likely to package a release than a commit

https://github.com/Haruno19/starfetch/issues/1

**Review by 06kellyjac — CHANGES_REQUESTED** [review]

**Comment by papojari** [comment]
currently blocked by
https://github.com/Haruno19/starfetch/issues/10

**Comment by papojari** [comment]
Everything should be fixed now. I patched the source code to have the nix store paths.

File: pkgs/tools/misc/starfetch/default.nix
Comment (06kellyjac) [inline pkgs/tools/misc/starfetch/default.nix:15]: You should use makeFlags instead of patching

https://github.com/Haruno19/starfetch/blob/main/makefile#L4

**Review by 06kellyjac — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/starfetch/default.nix
Comment (papojari) [reply pkgs/tools/misc/starfetch/default.nix:15]: What does makeFlags have to do with the paths inside the source code? If starship.cpp is hardcoded to /usr/local how is makeFlags gonna change that?

Comment (06kellyjac) [reply pkgs/tools/misc/starfetch/default.nix:15]: Ah its the cpp files.


If its the res dir it shouldn't be at the root of the $out

Comment (papojari) [reply pkgs/tools/misc/starfetch/default.nix:15]: I asked the developer about that too: https://github.com/Haruno19/starfetch/issues/2

Comment (papojari) [reply pkgs/tools/misc/starfetch/default.nix:15]: Should we tell him to put it somewhere in `/usr/share`?

Comment (06kellyjac) [reply pkgs/tools/misc/starfetch/default.nix:15]: :shrug: not sure. I haven't even looked deep enough to work out if these resources are static or should be updated etc so I couldn't give an accurate answer

Comment (Haruno19) [reply pkgs/tools/misc/starfetch/default.nix:15]: > Should we tell him to put it somewhere in `/usr/share`?

So, would it be better to use `/usr/share/starfetch` rather than `/usr/local/starfetch` to store resource files (txt and json files)? 
There's no problem in changing that, but may I ask why?

Comment (06kellyjac) [reply pkgs/tools/misc/starfetch/default.nix:15]: I'd imagine the argument is:

> /usr/share : Architecture-independent data

https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s11.html

Within `/usr/local` you could do `/usr/local/share/starfetch` but probably not `/usr/local/starfetch`

> No other directories, except those listed below, may be in /usr/local after first installing a FHS-compliant system.

see the directories section here

https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s09.html

So TL;DR `/usr/share/starfetch` or `/usr/local/share/starfetch` would be your options
And then I personally wouldn't add a `res` subdir since your res dir already is split nicely into 2 files and a dir

```
constellations/
help_message.txt
template
```

Comment (Haruno19) [reply pkgs/tools/misc/starfetch/default.nix:15]: Thank you for the exhaustive reply. 
I'll implement this changes as soon as possible, and then add a new release.

Comment (Haruno19) [reply pkgs/tools/misc/starfetch/default.nix:15]: [Release 0.0.2](https://github.com/Haruno19/starfetch/releases/tag/0.0.2) is now out. Files are now stored into `/usr/local/share/starfetch/` and the `res/` subdir has been removed.

**Comment by papojari** [comment]
I don't know if it's possible to use the makefile's install section for installing but I figured it doesn't make sense anyway because there are some commands in the makefile that aren't needed with Nix.

File: pkgs/tools/misc/starfetch/default.nix
Comment (06kellyjac) [inline pkgs/tools/misc/starfetch/default.nix:24]: This shouldn't just go in `$out/share`, it should go in `$out/share/starfetch`

**Pull request merged**
